distribution/integration/test-common/integration-test-utils/src/main/java/org/wso2/carbon/integration/common/utils/CarbonCommandToolsUtil.java

```
@@ -0,0 +1,412 @@
+                                                          "default credentials ", ex);
+        }
+    }
+
+    /**
+     * This method is to shutdown carbon server
+     *
+     * @param portOffset        - port offset
+     * @param automationContext - AutomationContext
+     * @throws CarbonToolsIntegrationTestException - Error occurred while shutdown the server
+     */
+    public static void serverShutdown(int portOffset,
+                                      AutomationContext automationContext)
+            throws CarbonToolsIntegrationTestException {
+        long time = System.currentTimeMillis() + DEFAULT_START_STOP_WAIT_MS;
+        log.info("Shutting down server..");
+        boolean logOutSuccess = false;
+        try {
+            if (ClientConnectionUtil.isPortOpen(
+                    Integer.parseInt(ExtensionConstants.SERVER_DEFAULT_HTTPS_PORT))) {
+
+                int httpsPort = Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset;
+                String url = automationContext.getContextUrls().getBackEndUrl();
+                String backendURL = url.replaceAll("(:\\d+)", ":" + httpsPort);
+
+                ServerAdminClient serverAdminServiceClient =
+                        new ServerAdminClient(backendURL, automationContext.getContextTenant().
+                                getTenantAdmin().getUserName(), automationContext.getContextTenant().
+                                getTenantAdmin().getPassword());
+
+                serverAdminServiceClient.shutdown();
+                while (System.currentTimeMillis() < time && !logOutSuccess) {
+                    logOutSuccess = isServerDown(automationContext, portOffset);
+                    // wait until server shutdown is completed
+                }
+                log.info("Server stopped successfully...");
+            }
+        } catch (Exception ex) {
```

Again make sure whether you need to catch all exceptions here ?